src/llama-chat.cpp

```diff
@@ -155,7 +158,7 @@ llm_chat_template llm_chat_detect_template(const std::string & tmpl) {
     } else if (tmpl_contains("[gMASK]sop")) {
         // chatglm3-6b
         return LLM_CHAT_TEMPLATE_CHATGML_3;
-    } else if (tmpl_contains("[gMASK]<sop>")) {
+    } else if (tmpl_contains("[gMASK]<sop>")) { /* old GLM4 models */
```

Comment:
It's better to simply move this code block to above the `else if (tmpl_contains("<|assistant|>") && tmpl_contains("<|user|>"))`, so the condition for GLM will have higher priority